lib/helper/TestCafe.js

```diff
@@ -135,37 +151,60 @@ class TestCafe extends Helper {
 
     this.iteration += 2; // Use different ports for each test run
     // @ts-ignore
-    this.testcafe = await createTestCafe('localhost', null, null);
+    this.testcafe = await createTestCafe('', null, null);
 
     this.debugSection('_before', 'Starting testcafe browser...');
 
     this.isRunning = true;
 
     // TODO Do we have to cleanup the runner?
     const runner = this.testcafe.createRunner();
-    runner
-      .src(this.dummyTestcafeFile)
-      .screenshots(global.output_dir, !this.options.disableScreenshots)
-      // .video(global.output_dir) // TODO Make this configurable
-      .browsers(this.options.show ? this.options.browser : `${this.options.browser}:headless`)
-      .reporter('minimal')
-      .run({
-        skipJsErrors: true,
-        skipUncaughtErrors: true,
-        quarantineMode: false,
-        // debugMode: true,
-        // debugOnFail: true,
-        // developmentMode: true,
-        pageLoadTimeout: this.options.getPageTimeout,
-        selectorTimeout: this.options.waitForTimeout,
-        assertionTimeout: this.options.waitForTimeout,
-        takeScreenshotsOnFails: true,
-      })
-      .catch((err) => {
-        this.debugSection('_before', `Error ${err.toString()}`);
-        this.isRunning = false;
-        this.testcafe.close();
+
+    if (this.options.browser !== 'remote') {
```

Comment:
Can you add tests for this changes?

Reply:
I am happy to do that but can you explain the current setup of the tests. It is not completely clear to me right now. Only unit and runner are executed during npm test right now. 
During CI builds following is executed:

test-unit 
test-rest 
test-acceptance.webdriverio 
test-acceptance.nightmare 
test-acceptance.puppeteer 
test-acceptance.protractor 

I don't see the helper tests are being invoked right now. I think I should place them there right? But how should I run / configure? 

Thanks in advance!

Reply:
@jvdieten you are right, we don't have much test for TestCafe and I'm really not so sure how can we set up testing for this PR. We have integration tests for TestCafe at `tests/helpers/TestCafe_test.js` but I don't see how it can be changed to test remote connections.

Reply:
@jvdieten you are right, we don't have much test for TestCafe and I'm really not so sure how can we set up testing for this PR. We have integration tests for TestCafe at tests/helpers/TestCafe_test.js but I don't see how it can be changed to test remote connections.